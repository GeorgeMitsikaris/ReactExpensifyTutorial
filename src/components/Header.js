import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <div>
        <header>
            <h1>Expensify</h1><br/>
            <NavLink to="/" exact={true} activeClassName="is-active">Dashboard</NavLink><br/>
            <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink><br/>
        </header>
    </div>
)

export default Header;