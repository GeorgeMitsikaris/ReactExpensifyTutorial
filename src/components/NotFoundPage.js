import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        <div>Page Not Found</div>
        <Link to="/">Go Home</Link>
    </div>
)

export default NotFoundPage;