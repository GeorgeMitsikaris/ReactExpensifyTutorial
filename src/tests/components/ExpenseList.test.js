import React from 'react';
import {shallow} from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import {expenses, expenseOne, expenseTwo, expenseThree} from '../fixtures/expenses';

test('should render ExpenseList with expenses', () => {
     const wrapper = shallow(<ExpenseList expenses = { expenses }/>);
     expect(wrapper).toMatchSnapshot(); 
})

test('should render ExpenseList with out expenses', () => {
     const wrapper = shallow(<ExpenseList expenses = { [] }/>); 
     expect(wrapper).toMatchSnapshot(); 
})