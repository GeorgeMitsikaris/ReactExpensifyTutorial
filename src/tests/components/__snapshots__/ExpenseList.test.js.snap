// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={10}
    createdAt={0}
    description="description for rent"
    id="1"
    key="1"
    note="rent"
  />
  <ExpenseListItem
    amount={100}
    createdAt={-345600000}
    description="description for gas"
    id="2"
    key="2"
    note="gas"
  />
  <ExpenseListItem
    amount={1000}
    createdAt={345600000}
    description="description for water"
    id="3"
    key="3"
    note="water"
  />
</div>
`;

exports[`should render ExpenseList with out expenses 1`] = `
<div>
  <p>
    No expenses found
  </p>
</div>
`;
